### PR TITLE
ZIOS-10131/ZIOS-10132: Implement timed message options in conversation details

### PR DESCRIPTION
**Pull request by alexaubry**
### Issues

This PR introduces the new interface for selecting enabling conversation-wide timed messages:

- [x] Cell in the conversation details
- [x] List for picking the timeout

### Solutions

- Introduce a cell type
- Group the "group" and "timed messages" into the same section controller

**Other changes**:

- Generic dequeuing of collection view cells
- Introduce a new base collection view cells with a separator (and use it as a cell superclass where needed)
- Allow settings a custom `messageDestructionTimeout` in `MockConversation`

## Dependencies

This needs to be released with these changes in data model/SE:

- [ ] Property for setting the conversation-wide message timeout
- [x] Message timeouts enum supporting custom values

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/GroupOptionsSectionController.swift [hunk 1]
```diff
@@ -18,51 +18,82 @@
...
+            let cell = collectionView.dequeueReusableCell(ofType: GroupDetailsGuestOptionsCell.self, for: indexPath)
+            cell.showSeparator = true
+            cell.isOn = conversation.allowGuests
+            cell.isUserInteractionEnabled = syncCompleted
+            cell.alpha = syncCompleted ? 1 : 0.48
+            return cell
+
+        case timeoutOptionsIndex:
+            let cell = collectionView.dequeueReusableCell(ofType: GroupDetailsTimeoutOptionsCell.self, for: indexPath)
+            cell.showSeparator = false
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/GroupDetails/Sections/GroupOptionsSectionController.swift]: Maybe some of the logic of both cases can be unified as well

Comment (alexaubry) [reply Wire-iOS/Sources/UserInterface/GroupDetails/Sections/GroupOptionsSectionController.swift]: Good call. I changed this in my latest commit 😄

**Comment by mikeger** [comment]
🔥  Conflicts after Swift 4

**Comment by mikeger** [comment]
@alexaubry should we merge it, so QA can automate the UI interactions?

**Comment by alexaubry** [comment]
@mikeger For now, this sets the local timer. If this is fine, I'll merge the PR and update the code later with the data model update.

**Comment by mikeger** [comment]
👍

**Pull request merged**